packages/codegen-ui-react/lib/__tests__/forms/validation.test.ts

```diff
@@ -297,4 +296,26 @@ describe('validateField tests', () => {
       errorMessage: 'test',
     });
   });
+
+  it('should validate Phone type', () => {
+    expect(validateField('kdj34324', [{ type: ValidationTypes.PHONE }])).toEqual({
+      hasError: true,
+      errorMessage: 'The value must be a valid phone number',
+    });
+
+    expect(validateField('2938493029', [{ type: ValidationTypes.PHONE, validationMessage: 'test' }])).toEqual({
+      hasError: false,
+      errorMessage: 'test',
+    });
+
+    expect(validateField('293 849 3029', [{ type: ValidationTypes.PHONE, validationMessage: 'test' }])).toEqual({
+      hasError: false,
+      errorMessage: 'test',
+    });
+
+    expect(validateField('293-849-3029', [{ type: ValidationTypes.PHONE, validationMessage: 'test' }])).toEqual({
+      hasError: false,
+      errorMessage: 'test',
+    });
+  });
```

Comment:
what if the number has 1 hyphen and 1 space? 123-123 1234

Reply:
Looks like DataStore will allow it.

https://github.com/aws-amplify/amplify-js/blob/46ee5dd91c61f49bad4da8286b2f97c737d96631/packages/datastore/src/util.ts#L675-L677

Rev'd to use this exact regex and added a test case for your point.

Reply:
Thanks!